scripts/us_census/pep_srh/process.py

```diff
@@ -0,0 +1,1015 @@
+    2010 - 2020     Processed As Is
+
+State
+    1980 - 1990     Processed As Is
+    1990 - 2000     Processed As Is
+    2000 - 2010     Aggregated from County to state
+                    (data matches with State Level files available)
+    2010 - 2020     Aggregated from County to state
+                    (data matches with State Level files available)
+
+National
+    1980 - 1990     Aggregted from State to National Level
+    1990 - 2000     Aggregted from State to National Level
+    2000 - 2010     Aggregted from State to National Level
+    2010 - 2020     Aggregted from State to National Level
+
+Also SV aggregation are produced while processing above files.
+E.g., Count_Person_White_HispanicOrLatino is calulated by addding
+Count_Person_Male_WhiteHispanicOrLatino and
+Count_Person_Male_WhiteHispanicOrLatino
+
+Before running this module, run download.sh script, it downloads required
+input files, creates necessary folders for processing.
+
+Folder information
+download_files - downloaded files (from US census website) are placed here
+process_files - intermediate processed files are placed in this folder.
+output_files - output files (mcf, tmcf and csv are written here)
+"""
+
+import os
+import pandas as pd
+
+from absl import app
+from absl import flags
+
+from constants import DOWNLOAD_DIR, PROCESS_AS_IS_DIR, PROCESS_AGG_DIR, RACE
+from constants import OUTPUT_DIR, STAT_VAR_COL_MAPPING, WORKING_DIRECTORIES
+from mcf_generator import generate_mcf
+from tmcf_generator import generate_tmcf
+
+_CODEDIR = os.path.dirname(os.path.realpath(__file__))
+_FLAGS = flags.FLAGS
+flags.DEFINE_string(
+    "data_directory", DOWNLOAD_DIR,
+    "Folder consisting of all input files required for processing. Run download.sh before running this python module"
+)
+
+# SR Columns with single race or combination with one or more race
+# These columns are not used as part of current import
+_SR_COLUMNS = [
+    'TOT_POP', 'TOT_MALE', 'TOT_FEMALE', 'WA_MALE', 'WA_FEMALE', 'BA_MALE',
+    'BA_FEMALE', 'IA_MALE', 'IA_FEMALE', 'AA_MALE', 'AA_FEMALE', 'NA_MALE',
+    'NA_FEMALE', 'TOM_MALE', 'TOM_FEMALE'
+]
+
+_SR_CMBN_COLUMNS = [
+    'WAC_MALE', 'WAC_FEMALE', 'BAC_MALE', 'BAC_FEMALE', 'IAC_MALE',
+    'IAC_FEMALE', 'AAC_MALE', 'AAC_FEMALE', 'NAC_MALE', 'NAC_FEMALE'
+]
+
+
+def _calculate_agg_measure_method(year, sv):
+    """
+    This method calculates measurement method of SV's.
+    US Census introduced additional Races and option to select more than one
+    Race in census survey from year 2000 onwards. 
+    This method will return measurement method as CensusPEPSurvey_RaceUpto1999
+    for data prior to 2000 and CensusPEPSurvey_Race2000Onwards for data from
+    2000 onwards. This is caluclated on those SV's only which has Race as one
+    of their property
+    
+    Args:
+      year: Year
+      SV: Statistical Variable
+    
+    Returns:
+      Measurement Method
+    """
+    for r in RACE:
+        if r in sv:
```

Comment:
may be safer to do `if sv.endswith(r):` since some races can be subset of others.
WhiteAlone is a substring of WhiteAlongOrInCombinationOf...

Reply:
I have made changes as suggested. As we were using following existing SV's - (Count_Person_WhiteAloneNotHispanicOrLatino, Count_Person_Male_WhiteAloneNotHispanicOrLatino, Count_Person_Female_WhiteAloneNotHispanicOrLatino). I have included these SV's in condition to handle this scenario as well.